### ISSUE TITLE
Ganache Next: Contracts can be deployed multiple times but Ganache will only show the latest

### ISSUE THREAD

**Issue by davidmurdoch** [trufflesuite/ganache#926]
Relevant code where we match transactions to contract creations: https://github.com/trufflesuite/ganache/commit/74b4e4ad9e3e15910992e086563314d21ef0e021#diff-bac743618c55c3d2f29a1e6d17275eb3R96

`contract.address` should probably be an array here (and the UI updated to reflect this)?

**Comment by seesemichaelj** [comment]
This is a UX question on whether or not it makes sense to be monitoring multiple instances of a contract within a Truffle project. I believe that Truffle projects currently assume it's only been deployed once

**Comment by davidmurdoch** [comment]
@DiscRiskandBisque and/or @shayzee. Any thoughts on @seesemichaelj's comment above about tracking multiple instances of a Truffle managed contract in Ganache?

**Comment by shayzee** [comment]
@davidmurdoch @seesemichaelj @DiscRiskandBisque yes, I agree about identifying a strong enough use case for us to add in the additional tracking feature. Is there a benefit to the user to be able to track and access multiple instances of a contract?

**Comment by seesemichaelj** [comment]
To add to my comment about "I'm not sure this is even a valid use case", Truffle's `truffle(-config)?.js` config file has a section called `networks` to say where the contract has been migrated to. It is currently not formatted for multiple addresses per network. See an example below:

```
  "networks": {
    "5777": {
      "events": {
        "0xa5263230ff6fc3abbbad333e24faf0f402b4e050b83a1d30ad4051f4e5d0f727": {
          "anonymous": false,
          "inputs": [
            {
              "indexed": false,
              "name": "greeting",
              "type": "string"
            }
          ],
          "name": "Greeting",
          "type": "event",
          "signature": "0xa5263230ff6fc3abbbad333e24faf0f402b4e050b83a1d30ad4051f4e5d0f727"
        }
      },
      "links": {},
      "address": "0xF90B560c46CF809D8B84f38B6E32ab5d4992A132",
      "transactionHash": "0x3167fd3652cd097325689744aa14ca7f1e30d2afac5926aa3c59be28ca0e00b3"
    }
  },
```

I don't think supporting this is viable unless the truffle framework supports this kind of workflow. Thoughts @shayzee (CC @gnidan for potential insight on the matter)?

**Comment by davidmurdoch** [comment]
Got some feedback from David Burela about this and he said he'd absolutely want multiple contract instances.

**Comment by shayzee** [comment]
Ok thanks for that feedback :)

**Comment by seesemichaelj** [comment]
This feature will not make the initial `2.0.0` release per discussions on Slack
